packages/server-core/src/projects/project/project.test.ts

```diff
@@ -25,89 +25,241 @@ Ethereal Engine. All Rights Reserved.
+            name: testProject.name
           },
-          params
+          getParams()
         )
-
-        let findParams = { ...params, query: { name: newProjectName } }
-        const project = (await app.service(projectPath).find(findParams)) as Paginated<ProjectType>
-        assert.strictEqual(project.data[0].name, newProjectName)
       })
+    })
+  })
 
-      it('should not add new project with same name as existing project', function () {
-        assert.rejects(async () => {
-          await app.service(projectPath).create(
-            {
-              name: newProjectName
-            },
-            params
-          )
-        })
-      })
+  describe('update', () => {
+    let sourceDirectory: string
+    let testUpdateProjectName: string
+
+    before(() => {
+      sourceDirectory = path.resolve(appRootPath.path, `packages/projects/projects/test-cloning-directory.git`)
+      fs.mkdirSync(sourceDirectory)
+      fs.writeFileSync(path.resolve(sourceDirectory, 'README.md'), 'Test Cloning Directory')
+
+      const git = useGit(sourceDirectory)
+      git.init()
+      git.add('README.md')
+      git.commit('initial commit')
     })
 
-    describe('remove', () => {
-      it('should remove project', async function () {
-        let findParams = { ...params, query: { name: newProjectName } }
-        const projectData = (await app.service(projectPath).find(findParams)) as Paginated<ProjectType>
-        await app.service(projectPath).remove(projectData.data[0].id, params)
-        const project = (await app.service(projectPath).find(findParams)) as Paginated<ProjectType>
-        assert.strictEqual(project.data.length, 0)
-      })
+    before(() => {
+      testUpdateProjectName = `test-update-project-name-${v1()}`
     })
 
     after(async () => {
-      await cleanup(app)
+      await cleanup(app, testUpdateProjectName)
+    })
+
+    after(() => {
+      fs.rmSync(sourceDirectory, { force: true, recursive: true })
+    })
+
+    it('should create and add the project details if it does not exist', async () => {
+      nock('https://api.github.com')
+        .get(/\/user.*/)
+        .reply(200, getAuthenticatedUser('test-user'))
+
+      await app.service(projectPath).update(
+        testProject.id,
+        {
+          sourceURL: sourceDirectory,
+          name: testUpdateProjectName
+        },
+        getParams()
+      )
+
+      const updatedProject = await app
+        .service(projectPath)
+        .find({ query: { name: testUpdateProjectName }, ...getParams() })
+      assert.ok(updatedProject.data[0].commitDate)
+      assert.equal(updatedProject.data[0].name, testUpdateProjectName)
+    })
+
+    it('should update the project details if it already exists', async () => {
+      nock('https://api.github.com')
+        .get(/\/user.*/)
+        .reply(200, getAuthenticatedUser('test-user'))
+
+      await app.service(projectPath).update(
+        testProject.id,
+        {
+          sourceURL: sourceDirectory,
```

Comment:
This is causing errors for me. The sourceURL is normally an https link to a GH repo, not a local file path. In project-helper.ts, in updateProject() around line 1468, we call getAuthenticatedRepo(), which lowercases the sourceURL, adds '.git' to the end, etc. Then, we call gitCloner.clone on that URL. 

For me, the path to sourceDirectory contains uppercase letters (/home/kyle/Development/Forks/etherealengine...), so when it tries to clone the output of getAuthenticatedRepo, the lowercasing of it means that it can't find the repo.